### DISCUSSION TITLE
No Browser Detected [ SUPABASE STORAGE ERROR ]

### DISCUSSION THREAD

**Discussion by rahul7400** [supabase/supabase#1731] [Questions] [answered]
hi there, here i am just trying to upload an image from nodejs by creating an nodejs api and i am getting the follwing error

> Error: No browser detected.
    at StorageFileApi.<anonymous> (C:\Users\Ritika\Desktop\test\node_modules\@supabase\storage-js\dist\main\lib\StorageFileApi.js:43:27)
    at Generator.next (<anonymous>)
    at C:\Users\Ritika\Desktop\test\node_modules\@supabase\storage-js\dist\main\lib\StorageFileApi.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\Ritika\Desktop\test\node_modules\@supabase\storage-js\dist\main\lib\StorageFileApi.js:4:12)
    at StorageFileApi.upload (C:\Users\Ritika\Desktop\test\node_modules\@supabase\storage-js\dist\main\lib\StorageFileApi.js:40:16)
    at upload (C:\Users\Ritika\Desktop\test\scripts\profileUpload.js:26:6)
    at C:\Users\Ritika\Desktop\test\scripts\profileUpload.js:8:9
    at IncomingForm.<anonymous> (C:\Users\Ritika\Desktop\test\node_modules\formidable\lib\incoming_form.js:107:9)
    at IncomingForm.emit (node:events:365:28)

I am just getting the image request from android app shown below:-

    var form = new formidable.IncomingForm()

    form.parse(req,function(err,fields,files){

        console.log(files)
        upload(files,supabase)
        
    })
   
Upoad funtion is below :-

   ` const {data,error} = await supabase
    .storage
    .from('profiles')
    .upload('public/avatar.jpg',files)
    console.log(error)`

When i am displaying the **files object** in console then it shows file details shown as below :-

> {
  file: File {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    size: 2112,
    path: 'C:\\Users\\Raghav\\AppData\\Local\\Temp\\upload_9cc0d2b17be91b2841c1be6d178fb830',
    name: '234.jpg',
    type: 'image/*',
    hash: null,
    lastModifiedDate: 2021-05-27T07:55:48.547Z,
    _writeStream: WriteStream {
      path: 'C:\\Users\\Raghav\\AppData\\Local\\Temp\\upload_9cc0d2b17be91b2841c1be6d178fb830',
      flags: 'w',
      mode: 438,
      fd: 3,
      start: undefined,
      pos: undefined,
      bytesWritten: 2112,
      closed: false,
      _writableState: [WritableState],
      _events: [Object: null prototype] {},
      _eventsCount: 0,
      _maxListeners: undefined,
      [Symbol(kFs)]: [Object],
      [Symbol(kIsPerformingIO)]: false,
      [Symbol(kCapture)]: false
    },
    [Symbol(kCapture)]: false
  }
}

I am actually not able to figure out the error so thats why i am writting in this discussion
**All the data which i was storing to database is getting stored correctly only uploading image to bucket is giving me error as mentioned above.
If you can help than please suggest me out about this error please**

**Comment by inian** [comment, marked as answer]
Hey @rahul7400, upload to Storage from node via supabase-js is [not supported yet.](https://github.com/supabase/storage-js/issues/5).  See [this discussion](https://github.com/supabase/supabase/discussions/1002) for more info.

**Reply by rahul7400** [reply]
Thank you for the response. I got this❤️ But is this going to take more time, if it is then i will shift to flutter.

**Reply by inian** [reply]
Hey Rahul, I will implement this in the next couple of weeks. You can use our Flutter client which has storage support too.

**Reply by rahul7400** [reply]
Thanks i ll switch to flutter then.👍
